### PR TITLE
Remove failure domain from AWSMachineSpec to respect CAPI's Machine/MD failure domain field

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind cleanup
/kind api-change

**What this PR does / why we need it**:
This PR removes the failure domain field from `AWSMachineSpec` to respect CAPI's Machine/MD's failure domain field.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes Part of #2355 

**Special notes for your reviewer**:

**Checklist**:
- [X] squashed commits
- [ ] includes documentation
- [X] adds unit tests
- [X] adds or updates e2e tests

### CODE REVIEW

**Comment by AverageMarcus** [comment]
Looks like this has accidentally pulled in the changes made in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3835 also. Other than that looks good.

**Comment by Ankitasw** [comment]
> Looks like this has accidentally pulled in the changes made in #3835 also. Other than that looks good.

Yes, it was intentional such that first #3835 gets merged, thats why this is on hold, but open for review 😄

**Comment by AverageMarcus** [comment]
Ah! I missed the hold.

File: api/v1beta1/conversion_test.go [hunk 1]
```diff
@@ -21,11 +21,53 @@ import (
...
+	return []interface{}{
+		AWSMachineFuzzer,
+		AWSMachineTemplateFuzzer,
+	}
+}
+
+func AWSMachineFuzzer(obj *AWSMachine, c fuzz.Continue) {
+	c.FuzzNoCustom(obj)
+	
+	// AWSMachine.Spec.FailureDomain, AWSMachine.Spec.Subnet.ARN and AWSMachine.Spec.AdditionalSecurityGroups.ARN has been removed in v1beta2, so setting it to nil in order to avoid v1beta1 --> v1beta2 --> v1beta1 round trip errors.
```
Comment (richardcase) [inline api/v1beta1/conversion_test.go:42]: I'm not sure setting them to nil is teh right way to do the v1beta1 --> v1beta2 --> v1beta1.

A user may have FailureDomains/ARN set in v1beta1, so i think the tests should store/restore the values. wdyt?

**Review by richardcase — COMMENTED** [review]
Looks good. One comment on the roundtrip fuzzing test.

Once the ARN change merges lets rebase this so we get rid of the ARN changes.

File: api/v1beta1/conversion_test.go (hunk 1 above)
Comment (richardcase) [reply api/v1beta1/conversion_test.go:42]: Doesn't feel right but it's what's done. I'd have though there was a pattern to preserve this. So we can resolve this.

**Comment by Skarlso** [comment]
/approve
/hold
until e2e

**Pull request merged**